front-end/src/app/shared/services/transaction.service.ts

```diff
@@ -77,6 +77,53 @@ export class TransactionService implements TableListService<Transaction> {
     return of(undefined);
   }
 
+  public getPreviousTransactionForCalendarYTD(
+    transaction: Transaction | undefined,
+    disbursement_date: Date | undefined,
+    dissemination_date: Date | undefined,
+    election_code: string | undefined,
+    candidate_office: string | undefined,
+    candidate_state: string | undefined,
+    candidate_district: string | undefined
+  ): Observable<Transaction | undefined> {
+    let actionDateString: string = this.datePipe.transform(disbursement_date, 'yyyy-MM-dd') ?? '';
+    if (actionDateString.length === 0) {
+      actionDateString = this.datePipe.transform(dissemination_date, 'yyyy-MM-dd') ?? '';
+    }
+
+    const transaction_id: string = transaction?.id ?? '';
+    const aggregation_group: AggregationGroups | undefined =
+      (transaction as ScheduleTransaction)?.aggregation_group ?? AggregationGroups.GENERAL;
+
+    if (transaction && actionDateString && candidate_office && aggregation_group && election_code) {
+      const params: { [key: string]: string } = {
+        transaction_id,
+        date: actionDateString,
+        aggregation_group,
+        election_code,
+        candidate_office,
+      };
+      if (candidate_state) {
```

Comment:
you can go ahead and just assign state and district in the params.  undefined will have the same behavior as not assigning them:

```
const params:  { [key: string]: string } = {
...
candidate_office,
candidate_state,
candidate_district
};
```

Reply:
This is actually in here to deal with a type error.  The API service's implementation of .get does not accept undefined values for parameters
![image](https://github.com/fecgov/fecfile-web-app/assets/104506225/7050e642-5029-43e5-9512-090f2f5e8c21)